sqlmesh/core/context.py

```diff
@@ -1261,6 +1261,8 @@ def render_dag(self, path: str, select_models: t.Optional[t.Collection[str]] = N
             path: filename to save the dag html to
             select_models: A list of model selection strings that should be included in the dag.
         """
+        if not path.endswith(".html"):
+            path += ".html"
```

Comment:
pathlib is useful for getting the file extension (they call it suffix for some reason):
```
assert pathlib.Path("example.txt") == ".txt"
```
So what you could do is something like this:
```
file_path = pathlib.Path(path)
suffix = pathlib.Path(file_path)
if suffix != ".html":
  logger.warning("Provided a file extension that is not `.html` but the output is an `.html` file. Updating to be an html file.")
  path = str(file_path.with_suffix(".html"))
```

This way you don't end up with something like `file.txt.html`.